tensorboard/pip_package/build_pip_package.sh

```diff
@@ -94,140 +83,39 @@ cp -LR "${RUNFILES}/org_tensorflow_serving_api/tensorflow_serving" tensorboard/_
-  mkfifo pipe
-  tensorboard --port=0 --logdir=smokedir 2>pipe &
-  perl -ne 'print STDERR;/http:.*:(\d+)/ and print $1.v10 and exit 0' <pipe >port
-  curl -fs "http://localhost:$(cat port)" >index.html
-  grep '<tf-tensorboard' index.html
-  curl -fs "http://localhost:$(cat port)/data/logdir" >logdir.json
-  grep 'smokedir' logdir.json
-  curl -fs "http://localhost:$(cat port)/data/plugin/projector/runs" >projector_runs.json
-  # logdir does not contain any checkpoints and thus an empty runs.
-  grep '\[\]' projector_runs.json
-  curl -fs "http://localhost:$(cat port)/data/plugin/projector/projector_binary.html" >projector_binary.html
-  grep '<vz-projector-dashboard' projector_binary.html
-  kill $!
-
-  # Test TensorBoard APIs
-  export TF_CPP_MIN_LOG_LEVEL=1  # Suppress spammy TF startup logging.
-  python -c "
-import tensorboard as tb
-assert tb.__version__ == tb.version.VERSION
-from tensorboard.plugins.projector import visualize_embeddings
-tb.notebook.start  # don't invoke; just check existence
-from tensorboard.plugins.hparams import summary_v2 as hp
-hp.hparams_pb({'optimizer': 'adam', 'learning_rate': 0.02})
-"
-  if [ -n "${smoke_tf}" ]; then
-    # Only test summary scalar, beholder, and mesh summary
-    python -c "
-import tensorboard as tb
-tb.summary.scalar_pb('test', 42)
-from tensorboard.plugins.beholder import Beholder, BeholderHook
-from tensorboard.plugins.mesh import summary
-"
-  fi
-
-  if [ -n "${smoke_tf}" ]; then
-    # Exhaustively test various sequences of importing tf.summary.
-    test_tf_summary() {
-      # First argument is subpath to test, e.g. '' or '.compat.v2'.
-      import_attr="import tensorflow as tf; a = tf${1}.summary; a.write; a.scalar"
-      import_as="import tensorflow${1}.summary as b; b.write; b.scalar"
-      import_from="from tensorflow${1} import summary as c; c.write; c.scalar"
-      printf '%s\n' "${import_attr}" "${import_as}" "${import_from}" | python -
-      printf '%s\n' "${import_attr}" "${import_from}" "${import_as}" | python -
-      printf '%s\n' "${import_as}" "${import_attr}" "${import_from}" | python -
-      printf '%s\n' "${import_as}" "${import_from}" "${import_attr}" | python -
-      printf '%s\n' "${import_from}" "${import_attr}" "${import_as}" | python -
-      printf '%s\n' "${import_from}" "${import_as}" "${import_attr}" | python -
-    }
-    test_tf_summary '.compat.v2'
-    is_tf_2() {
-      python -c "import tensorflow as tf; assert tf.__version__[:2] == '2.'" \
-        >/dev/null 2>&1
-    }
-    if is_tf_2 ; then
-      test_tf_summary ''
-    fi
-  fi
-
-  deactivate
-  cd ..
-  rm -rf "${smoke_venv}"
-}
-
-case "${smoke}" in
-  "all")
-    smoke 2 "${tf_version}"
-    smoke 3 "${tf_version}"
-    ;;
-  "1")
-    # Empty string indicates to use the default "python".
-    smoke "" "${tf_version}"
-    ;;
-  "0")
-    printf "\nSkipping smoke test\n\n"
+cd "${original_wd}"  # Bazel gives "${output}" as a relative path >_>
```

Comment:
Can we just resolve `output` to an absolute path while we're still in the original working directory, rather than having to `cd` back to it later?

Reply:
I don’t know a portable way to do that. GNU `readlink -e`/`-f` works
perfectly on Linux, but on macOS `readlink`(1) is basically just
`readlink`(2). `realpath`(1) is also not POSIX. ([Here is a handy list
of portable utilities][posix]—it’s not large, especially when you look
at (e.g.) `xargs` and see that there’s no portable `-0` option!)

[posix]: http://pubs.opengroup.org/onlinepubs/9699919799/idx/utilities.html

I mean, there’s always

```shell
path="$(python -c 'print(__import__("os").path.realpath(__import__("sys").argv[1]))' "${path}")"
```

which has the downside of being a bit hideous (and also not correct
because command substitutions strip whitespace); just `cd`ing back is
significantly simpler imho.

Reply:
(If there _is_ a portable way to do this, I would love to know!)

Reply:
What about

```sh
case "${output}" in 
  /*) ;;
  *) output="${PWD}/${output}" ;;
esac
```